core/src/main/java/org/apache/iceberg/jdbc/JdbcTable.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.jdbc;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JdbcTable {
```

Comment:
I'm not sure that I think this class is valuable. It is nice that it keeps track of the SQL statements and exposes some high-level methods, like `update`. The problem is that this provides many of the same operations as catalog, just under different names and with a slightly different API. For example, `getAll(Namespace)` is basically the same thing as `Catalog.listTables`.

This class is also used in inconsistent ways. In `Catalog.delete` and in `TableOperations`, it is used for a single table even though it can work with any table and has no state other than catalog and connection. It is also used in `Catalog.listTables` and `Catalog.renameTable`, even though there is no table to encapsulate.

I think that the problem is that this is attempting to encapsulate a table, but it is really more about abstracting some of the SQL details. I think an improvement would be to instantiate one per catalog instead (and pass in a connection pool so it can be shared with `TableOperations`). You'd probably want to rename it to something like `TableSQL`?

Reply:
One more thing: it would be helpful if all of the methods in this class that correspond to methods in `Catalog` were named the same way as the ones in `Catalog`. For example, `getAll` is really `listTables` and `updateTableName` is really `renameTable`. Using names consistently will make it easier for people to understand what this is doing.

Reply:
this was good idea implement it, and reorganized code a bit 
>  I think an improvement would be to instantiate one per catalog instead (and pass in a connection pool so it can be shared with TableOperations). You'd probably want to rename it to something like TableSQL?

Reply:
> I'm not sure that I think this class is valuable. ...

i see the pint totally makes sense. i see two options to address it 
- merge it to `Catalog` and `TableOperations` class and only pass connection pool to `TableOperations`, similar to hive.
- split it to `CatalogSQL`, `TableOperationsSQL` classes. `TableOperationsSQL` is used by `TableOperations`, and it additionally gets table identifier with constructor

maybe its best to merge it to Catalog and address it with the future changes?

Reply:
went with first option and moved code to to `Catalog` and `TableOperations` classes.